astropy/config/paths.py

```diff
@@ -144,7 +118,11 @@ def get_cache_dir(rootname: str = "astropy") -> str:
 
 class _SetTempPath:
     _temp_path: Path | None = None
-    _default_path_getter: Callable[[str], str]
+
+    # This base class serves as a deduplication layer for its only two intended
+    # children (set_temp_cache and set_temp_config)
+    _directory_type: Literal["cache", "config"]
+    _directory_env_var: str
```

Comment:
`_directory_env_var` should be annotated using `Literal` because it can only hold a small set of values which we are not free to choose because they are defined by an external standard.

Reply:
I'm not satisfied with this suggestion because it leaves room for 2 invalid cases, namely

```python
    _directory_typ = "cache"
    _directory_env_var = "XDG_CONFIG_HOME"
```
and
```python
    _directory_typ = "config"
    _directory_env_var = "XDG_CACHE_HOME"
```

So I refactored this again into an enum. Now, the pairing is expressed in the annotation.

Reply:
update: my enum refactor was broken beyond repair. Welp, maybe I'll avoid chasing perfection here.

Reply:
I still think that it's simpler to check that these two attributes that are right next to each other agree when reading the code than it would be to understand some clever mechanism that would generate one from the other at runtime.

Reply:
Agreed. I'm still slightly frustrated by the outcome but I can certainly live with it !